addons/hr_holidays/models/hr_employee_base.py

```diff
@@ -216,6 +216,8 @@ def _get_consumed_leaves(self, leave_types, target_date=False, ignore_future=Fal
                             # to give a warning if the total exceeds what will be accrued.
                             if allocation.date_from > leave.date_to.date() or (allocation.date_to and allocation.date_to < leave.date_from.date()):
                                 continue
+                            if 'allocation_refuse_check' in self.env.context and leave.create_date <= allocation.create_date and leave.date_from.date() > target_date:
+                                continue
```

Comment:
This is a weird condition IMO; the create date makes no real sense since you could also:
- create 2 allocations both making a leave valid with different end dates
- create a leave valid on both allocations
- try to refuse the first allocation to expire

Additionally, this method is only for information purposes, I wouldn't start to add conditions and other stuff in here. Everything should be handled in the allocation